### PR TITLE
Add return shape for Throwable::getTrace

### PR DESCRIPTION
**Pull request by ciaranmcnulty**
This is based on the documentation for debug-backtrace:
   https://php.net/debug_backtrace

Experimentation that shows the `object` field is not populated for exceptions:
   https://3v4l.org/iQoni

And validation on psalm.dev:
   https://psalm.dev/r/b35a8df2f3

Should I also update the stubs?
What tests should I add?

### CODE REVIEW

**Comment by ciaranmcnulty** [comment]
Updated to allow optional `args` (in the case of an included file)

https://psalm.dev/r/11247acbc9

**Comment by orklah** [comment]
Seems great!

You'll have to change Callmap_Historical too for tests to pass

**Comment by ciaranmcnulty** [comment]
@orklah should I update /stubs too? I'm not sure the connection between the two things

**Comment by AndrolGenhald** [comment]
Yes, stubs should be updated too. CallMap is there because it's much faster to parse for simple functions, but stubs exist to add annotations that don't work in the CallMap, so the stubs override the CallMap.

**Review by orklah — APPROVED** [review]

**Comment by ciaranmcnulty** [comment]
I rebased onto 4.x

I'm not sure what's happened there with Appveyor?

**Comment by ciaranmcnulty** [comment]
Closed/reopened to try and nudge appveyor :/

**Comment by orklah** [comment]
Appveyor was removed from our CI ages ago, but sometimes on rebase it pops again, not sure why. As long as the rest of the CI runs, it's good

**Comment by orklah** [comment]
Thanks!

File: dictionaries/CallMap.php [hunk 1]
```diff
@@ -346,7 +346,7 @@
 'ArgumentCountError::getLine' => ['int'],
 'ArgumentCountError::getMessage' => ['string'],
 'ArgumentCountError::getPrevious' => ['?Throwable'],
-'ArgumentCountError::getTrace' => ['list<array<string,mixed>>'],
+'ArgumentCountError::getTrace' => ['list<array{file:string,line:int,function:string,class?:string,type?:\'::\'|\'->\',args?:array<mixed>}>'],
```
Comment (staabm) [inline dictionaries/CallMap.php:349]: per https://github.com/phpstan/phpstan-src/pull/914#issuecomment-1007370554 `file` needs to be nullable

Comment (staabm) [reply dictionaries/CallMap.php:349]: and `class` can be narrowe to `class-string`

Comment (AndrolGenhald) [reply dictionaries/CallMap.php:349]: `line` is optional as well apparently.

**Pull request merged**